### ISSUE TITLE
Unexpected error raised for table marked as presentation

### ISSUE THREAD

**Issue by scottaohara** [dequelabs/axe-core#3654]
### Product

axe-core

### Product Version

4.4.2

### Latest Version

- [X] I have tested the issue with the latest version of the product

### Issue Description

The following code snippet was flagged as a serious error with axe for use of the headers attribute.  

```
<table role=presentation>
 <tr><th hidden>...</th><th>...</th></tr>
  <tr>
     <td headers=foo hidden>...</td>
     <td>...</td>
  </tr>
</table>
```

However, this doesn't make any sense to flag as an error because:
1. the table has been marked as role=presentation so table semantics are kicked to the curb anyway.
2. the cell with the erroneous `headers` attribute on it also has the `hidden` attribute, so even if the table was still exposed as a table, the cell is hidden so still can't be an accessibility issue.

**Comment by padmavemulapati** [comment]
Valdiated with the latest axe-core develop branch code base, 
in the test script table has been marked as role=presentation  and td atrribute as hidden so test snippet should not fail for `td-headers-attr` rule
```
<table role=presentation>
 <tr><th hidden>...</th><th>...</th></tr>
  <tr>
     <td headers=foo hidden>...</td>
     <td>...</td>
  </tr>
</table>

```



![Image](https://user-images.githubusercontent.com/47409579/192967378-5e882652-a341-4108-8afa-f8425ee7b3e3.png)
